compiler/rustc_hir/src/hir.rs

```diff
@@ -3491,17 +3489,20 @@ pub enum Node<'hir> {
     Arm(&'hir Arm<'hir>),
     Block(&'hir Block<'hir>),
     Local(&'hir Local<'hir>),
-
     /// `Ctor` refers to the constructor of an enum variant or struct. Only tuple or unit variants
     /// with synthesized constructors.
     Ctor(&'hir VariantData<'hir>),
-
     Lifetime(&'hir Lifetime),
     GenericParam(&'hir GenericParam<'hir>),
-
     Crate(&'hir Mod<'hir>),
-
     Infer(&'hir InferArg),
+    WhereBoundPredicate(&'hir WhereBoundPredicate<'hir>),
+    Let(&'hir Let<'hir>),
+    // Span by reference to minimize `Node`'s size
+    #[allow(rustc::pass_by_value)]
+    ArrayLenInfer(&'hir Span),
+    #[allow(rustc::pass_by_value)]
+    Err(&'hir Span),
```

Comment:
Do we really need to store that span? Could we return `DUMMY_SP` when needed?

Reply:
We could return `DUMMY_SP`, yes.
`index_hir` just has a close enough real span that can be used, so I used it (that makes `Node::span` always return real spans).